### ISSUE TITLE
Slim Train API doesn't provide control to user to extract loss and accuracy metrics

### ISSUE THREAD

**Issue by deepak-2017** [tensorflow/tensorflow#18453]
I am referring to [train_image_classifier.py][1] script provided by Tensorflow for image classification.

Script uses slim.learning.train API to kick off the training and then it takes care for everything internally.
Though it logs the metrics i.e. loss/accuracy but that's all internal to slim. I doesn't provide handle to get periodic updates of loss/accuracy in my script while training is going on.
My requirement is to extract loss and accuracy metrics after every n number of epochs, i.e. 5 and dump that into database for later analysis.

But because of high level abstraction of slim train API, It seems there is now way to get periodic updates loss/accuracy info during the training.

Is it possible to tap the training process and get the loss/accuracy values after every given number of steps/epochs ?
If it's possible, please share the pseudo code for that.

It will be really helpful for me. 

Thanks.

  [1]: https://github.com/tensorflow/models/blob/master/research/slim/train_image_classifier.py

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by deepak-2017** [comment]
Thanks your reply.
Adding the system info here.

System information
------------------

 - Have I written custom code - Yes
 - OS Platform CentOS - 7.2.1511
 - TensorFlow installed from - Binary
 - TensorFlow version - 1.3.0
 - Python version - 2.7
 - Bazel version - N/A
 - CUDA/cuDNN version - N/A
 - GPU model and memory - N/A

**Comment by deepak-2017** [comment]
@rohan100jain  - Kindly help me with your answer.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 61 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 76 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 91 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rohan100jain: It has been 106 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rohan100jain** [comment]
Doesn't the log_every_n_steps flag give you what you need? It'll print out metrics periodically?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 44 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by harshini-gadige** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
